src/Integration.UnitTests/Suppression/SonarQubeIssuesProviderTests.cs

```diff
@@ -398,6 +393,94 @@ public void GetIssues_UnboundSolution_NoErrors()
             matches.Should().BeEmpty();
         }
 
+        [TestMethod]
+        public void GetIssues_ErrorInInitialFetchTask_IsSuppressed()
+        {
+            var issue1 = new SonarQubeIssue("folder1/file1", "hash1", 0, "message", "sqkey:sqkey:projectID1",
+                SonarQubeIssueResolutionState.FalsePositive, "S101");
+
+            EventWaitHandle waitHandle = new EventWaitHandle(false, EventResetMode.ManualReset);
+
+            Func<IList<SonarQubeIssue>> serviceFetchIssuesTask = () =>
+            {
+                waitHandle.Set(); // signal so the test can continue
+                throw new ApplicationException("dummy error from mock");
+            };
+
+            SetupSolutionBinding(isBound: true, isConnected: true, projectKey: "sqkey", issues: new List<SonarQubeIssue> { issue1 });
+            mockSqService.Setup(x => x.GetSuppressedIssuesAsync(It.IsAny<string>(), It.IsAny<CancellationToken>()))
+                .ReturnsAsync(serviceFetchIssuesTask)
+                .Verifiable();
+
+            // 1. Create the issue provider
+            // The initial fetch should be triggered, but not yet completed
+            var issuesProvider = new SonarQubeIssuesProvider(mockSqService.Object, mockTracker.Object, mockTimerFactory.Object);
+
+            var waitSignaled = waitHandle.WaitOne(Debugger.IsAttached ? 20000 : 5000); // wait for fetch to start...
+            waitSignaled.Should().BeTrue(); // error - fetch has not started running
+
+            // 2. Now request the issues - task completes with an error
+            var matches = issuesProvider.GetSuppressedIssues("projectid1", "folder1/file1");
+
+            VerifyServiceGetIssues(Times.Once());
+            matches.Should().BeEmpty();
+
+            // 3. Now fetch again - should not wait again, should not error
+            matches = issuesProvider.GetSuppressedIssues("folder1/file1", "projectid1");
+            matches.Should().BeEmpty();
+            VerifyServiceGetIssues(Times.Once());
+        }
+
+        [TestMethod]
+        public void GetIssues_ErrorFetchingOnBindingChanged_IsSuppressed()
+        {
+            ExecuteGetIssuesOnTriggerWithError("projectId", () => RaiseSolutionBoundEvent(true, "projectId"));
+            VerifyTimerStart(Times.Exactly(2)); // once on construction, once in the refresh 
```

Comment:
Previously the timer would not be started if an exception occurred during _SynchronizeSuppressedIssues_.